gms/version_generator.hh

```diff
@@ -17,6 +17,13 @@ namespace gms {
 using generation_type = utils::tagged_integral<struct generation_type_tag, int32_t>;
 using version_type = utils::tagged_integral<struct version_type_tag, int32_t>;
 
+void validate_gossip_generation(int64_t generation_number);
+inline void debug_validate_gossip_generation([[maybe_unused]] int64_t generation_number) {
+#ifndef SCYLLA_BUILD_MODE_RELEASE
+    validate_gossip_generation(generation_number);
+#endif
```

Comment:
Is gossip really so performance sensitive that we can't do this in release mode?

Reply:
I see, makes sense.